leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -153,31 +164,33 @@ private PaxosQuorumChecker() {
 
     /**
      * Collects a list of responses from remote services.
-     * This method may short-circuit if a quorum can no longer be obtained (depending on the
-     * shortcircuitIfQuorumImpossible parameter) and cancels pending requests once a quorum has been obtained.
+     * This method may short-circuit depending on the {@code shouldSkipNextRequest} predicate parameter and cancels
```

Comment:
nit: I'd name the param `shouldSkipRemainingRequests` or even just `shouldShortcircuit` or something, since it does a lot more than skip the next request...

Reply:
will do, added to #4112